File: z-clients/java/src/main/java/com/svix/diom/apis/RateLimitNamespace.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.ApiException;
import com.svix.diom.HttpClient;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.RateLimitConfigureNamespaceIn;
import com.svix.diom.models.RateLimitConfigureNamespaceOut;
import com.svix.diom.models.RateLimitGetNamespaceIn;
import com.svix.diom.models.RateLimitGetNamespaceOut;

public class RateLimitNamespace {
    private final HttpClient client;

    public RateLimitNamespace(HttpClient client) {
        this.client = client;
    }

    /** Configure rate limiter namespace */
    public RateLimitConfigureNamespaceOut configure(
        final RateLimitConfigureNamespaceIn rateLimitConfigureNamespaceIn
    ) throws IOException, ApiException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.rate-limit.namespace.configure",
            null,
            rateLimitConfigureNamespaceIn,
            RateLimitConfigureNamespaceOut.class
        );
    }

    /** Get rate limiter namespace */
    public RateLimitGetNamespaceOut get(
        final RateLimitGetNamespaceIn rateLimitGetNamespaceIn
    ) throws IOException, ApiException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.rate-limit.namespace.get",
            null,
            rateLimitGetNamespaceIn,
            RateLimitGetNamespaceOut.class
        );
    }
}

File: z-clients/python/diom/apis/rate_limit_namespace.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    RateLimitConfigureNamespaceIn,
    RateLimitConfigureNamespaceOut,
    RateLimitGetNamespaceIn,
    RateLimitGetNamespaceOut,
)


class RateLimitNamespaceAsync(ApiBase):
    async def configure(
        self,
        rate_limit_configure_namespace_in: RateLimitConfigureNamespaceIn,
    ) -> RateLimitConfigureNamespaceOut:
        """Configure rate limiter namespace"""
        body = rate_limit_configure_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.rate-limit.namespace.configure",
            body=body,
            response_type=RateLimitConfigureNamespaceOut,
        )

    async def get(
        self,
        rate_limit_get_namespace_in: RateLimitGetNamespaceIn,
    ) -> RateLimitGetNamespaceOut:
        """Get rate limiter namespace"""
        body = rate_limit_get_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.rate-limit.namespace.get",
            body=body,
            response_type=RateLimitGetNamespaceOut,
        )


class RateLimitNamespace(ApiBase):
    def configure(
        self,
        rate_limit_configure_namespace_in: RateLimitConfigureNamespaceIn,
    ) -> RateLimitConfigureNamespaceOut:
        """Configure rate limiter namespace"""
        body = rate_limit_configure_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.rate-limit.namespace.configure",
            body=body,
            response_type=RateLimitConfigureNamespaceOut,
        )

    def get(
        self,
        rate_limit_get_namespace_in: RateLimitGetNamespaceIn,
    ) -> RateLimitGetNamespaceOut:
        """Get rate limiter namespace"""
        body = rate_limit_get_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.rate-limit.namespace.get",
            body=body,
            response_type=RateLimitGetNamespaceOut,
        )
